add can fd support to can transport layer

fgeyer opened the pull request.

PlumIsNoGeek committed:
fix platform independent path

PlumIsNoGeek committed:
add canfd support for physical interfaces

PlumIsNoGeek committed:
increment version to 1.2.0

fgeyer committed:
Merge pull request #1 from fgeyer/can-fd-support

Can fd support

richClubb commented:
Cheers for this. I'll have a review and get it in this week.

fgeyer commented:
Any news about your review?

PlumIsNoGeek committed:
fix app_name parameter for CanConnectionFactory

richClubb commented on uds/uds_communications/TransportProtocols/Can/CanTp.py:
```diff
@@ -279,7 +286,7 @@ def send(self, payload, functionalReq=False):
             if(timeoutTimer.isExpired()):
                 raise Exception("Timeout waiting for message")
 
-            sleep(0.001)
+            #sleep(0.01)
 
```
Why has this been changed? Is it important? The sleep was in to mitigate intermittent issues with CanTp correctly processing the responses

richClubb commented on MANIFEST.in:
```diff
@@ -1,5 +1,5 @@
-include uds\uds_communications\TransportProtocols\Can\config.ini
-include uds\uds_communications\TransportProtocols\Lin\config.ini
\ No newline at end of file
+include uds/uds_configuration/defaultConfig.ini
+include uds/uds_communications/Uds/config.ini
+include uds/uds_communications/TransportProtocols/config.ini
+include uds/uds_communications/TransportProtocols/Can/config.ini
+include uds/uds_communications/TransportProtocols/Lin/config.ini
```
Did you change this manually? Or was it automatic?

richClubb reviewed (commented):
Mostly good :) two comments that need an answer before I can merge it in. Sorry it took so long, and thanks for contributing this.

Which devices have you tested this with? Vector and Peak?

PlumIsNoGeek committed:
add cast

PlumIsNoGeek committed:
add connection shutdown to free can channel

fbekhechi committed:
- Parse Data from ODX file of type Integer into an array of bytes based on data bit length instead of integer type.

- Negative response returns dictionary with NRC value instead of throwing exception.
- Set can message dlc to its length (the bus DLC is already handled in python-can).
- Update unit tests for the function containers.

PlumIsNoGeek committed:
readd sleep which is better for performance

fbekhechi committed:
- Shutdown CAN connection only if not created externally.

- Correct bug for external can vector bus connection.

fbekhechi committed:
Merge branch 'master' of https://github.com/fgeyer/python-uds

PlumIsNoGeek committed:
add vector serial auto detection

PlumIsNoGeek committed:
fix key check

PlumIsNoGeek committed:
fix deprecation warning

fbekhechi committed:
- Fix min payload single frame bug.

fbekhechi committed:
- Add property to change reqId and resId

Commit:
add vector serial auto detection

fgeyer committed:
Update version

fbekhechi committed:
- reset CanConnectionFactory at can disconnect

mol4rt committed:
Refactored CanTp.py and UdsConfigTool.py for usage of uds auxiliary

mol4rt committed:
Refactored CanTp.py and UdsConfigTool.py for usage of uds auxiliary

fgeyer committed:
Merge pull request #2 from fgeyer/TMSEA3-1232-itf-refactor-can-uds

Refactored CanTp.py and UdsConfigTool.py for usage of uds auxiliary

fgeyer committed:
Revert "Refactored CanTp.py and UdsConfigTool.py for usage of uds auxiliary"

fgeyer committed:
Merge pull request #3 from fgeyer/revert-2-TMSEA3-1232-itf-refactor-can-uds

Revert "Refactored CanTp.py and UdsConfigTool.py for usage of uds auxiliary"

Commit:
prepare 1.2.2

fgeyer committed:
Tmsea3 1232 itf refactor can uds (#5)

* Refactored CanTp.py and UdsConfigTool.py for usage of uds auxiliary
* add parametrized wait time to iso tp/uds send

Commit:
fix typo

Commit:
fix return types

Commit:
readd lost lines

Commit:
fix isotp

Commit:
fix wait

Commit:
fix thread overload while waiting for uds response
